x-pack/plugins/security_solution/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -211,6 +214,49 @@ export const signalRulesAlertType = ({
           await ruleStatusService.error(errorString);
           wroteStatus = true;
         }
+        const timestampFieldCaps = await services.scopedClusterClient.fieldCaps({
+          index,
+          fields: timestampFields,
+          allow_no_indices: false,
+          include_unmapped: true,
+        });
+
+        if (
+          !wroteStatus &&
+          params.timestampOverride != null &&
+          (isEmpty(timestampFieldCaps?.body?.fields) ||
+            timestampFieldCaps?.body?.fields[params.timestampOverride] == null ||
+            timestampFieldCaps?.body?.fields[params.timestampOverride]?.unmapped?.indices != null)
+        ) {
+          // if there is a timestamp override and the unmapped array for the timestamp override key is not empty,
+          // partial failure
+          const errorString = `The following indices are missing the timestamp override field "${
+            params.timestampOverride
+          }": ${JSON.stringify(
+            isEmpty(timestampFieldCaps?.body?.fields)
+              ? timestampFieldCaps.body.indices
+              : timestampFieldCaps?.body?.fields[params.timestampOverride]?.unmapped?.indices
+          )}`;
+          logger.debug(buildRuleMessage(errorString));
+          await ruleStatusService.partialFailure(errorString);
+          wroteStatus = true;
+        } else if (
+          !wroteStatus &&
+          params.timestampOverride == null &&
+          (isEmpty(timestampFieldCaps?.body?.fields) ||
+            timestampFieldCaps?.body?.fields['@timestamp'].unmapped?.indices != null)
+        ) {
+          // if there is no timestamp override and the unmapped array is not empty,
+          // partial failure
+          const errorString = `The following indices are missing the timestamp field "@timestamp": ${JSON.stringify(
+            isEmpty(timestampFieldCaps?.body?.fields)
+              ? timestampFieldCaps.body.indices
+              : timestampFieldCaps?.body?.fields['@timestamp']?.unmapped?.indices
+          )}`;
+          logger.debug(buildRuleMessage(errorString));
```

Comment:
`logger.debug` that looks like it should be `logger.error` here and these chunks of code could really be pulled out into functions that take the '@timestamp' or override as a param for a builder. Would read better and we wouldn't have to worry about two places to change something in regards to building error messages.

Reply:
My thinking was use logger.debug since this is a partial failure and not an unhandled error case. I can change it to logger.error though.